File: basico/Python/NumeroComplexo.py
class NumeroComplexo:

    def __init__(self, parteReal, parteImaginaria):
        self.__parteReal = parteReal
        self.__parteImaginaria = parteImaginaria

    def somar(self, outroComplexo):
        self.__parteReal += outroComplexo.getParteReal()
        self.__parteImaginaria += outroComplexo.getParteImaginaria()

    def subtrair(self, outroComplexo):
        self.__parteReal -= outroComplexo.getParteReal()
        self.__parteImaginaria -= outroComplexo.getParteImaginaria()

    def imprimir(self):
        if self.__parteImaginaria >= 0:
            print(f"{self.__parteReal} + i{self.__parteImaginaria}")
        else:
            print(f"{self.__parteReal} - i{abs(self.__parteImaginaria)}")



    def getParteReal(self):
        return self.__parteReal

    def setParteReal(self, parteReal):
        self.__parteReal = parteReal

    def getParteImaginaria(self):
        return self.__parteImaginaria

    def setParteImaginaria(self, parteImaginaria):
        self.__parteImaginaria = parteImaginaria

File: basico/Java/NumeroComplexo.java
public class NumeroComplexo {
        private double parteReal;
        private double parteImaginaria;

        public NumeroComplexo(double parteReal, double parteImaginaria) {
            this.parteReal = parteReal;
            this.parteImaginaria = parteImaginaria;
        }
        public double getParteReal() {
            return parteReal;
        }
        public void setParteReal(double parteReal) {
            this.parteReal = parteReal;
        }
        public double getParteImaginaria() {
            return parteImaginaria;
        }
        public void setParteImaginaria(double parteImaginaria) {
            this.parteImaginaria = parteImaginaria;
        }
        public void somar(NumeroComplexo outro) {
            this.parteReal = this.parteReal + outro.getParteReal();
            this.parteImaginaria = this.parteImaginaria + outro.getParteImaginaria();
        }
        public void subtrair(NumeroComplexo outro) {
            this.parteReal = this.parteReal - outro.getParteReal();
            this.parteImaginaria = this.parteImaginaria - outro.getParteImaginaria();
        }
        public void imprimir() {
            if (this.parteImaginaria >= 0) {
                System.out.println(this.parteReal + " + i" + this.parteImaginaria);
            } else {
                System.out.println(this.parteReal + " - i" + Math.abs(this.parteImaginaria));
            }
        }
    }

